internal/services/hsm/cloud_hardware_security_module_cluster_resource.go

```diff
@@ -0,0 +1,483 @@
+	Status          string `tfschema:"status"`
+	Description     string `tfschema:"description"`
+	ActionsRequired string `tfschema:"actions_required"`
+}
+
+var _ sdk.ResourceWithUpdate = CloudHardwareSecurityModuleClusterResource{}
+
+type CloudHardwareSecurityModuleClusterResource struct{}
+
+func (CloudHardwareSecurityModuleClusterResource) ResourceType() string {
+	return "azurerm_cloud_hardware_security_module_cluster"
+}
+
+func (CloudHardwareSecurityModuleClusterResource) ModelObject() interface{} {
+	return &CloudHardwareSecurityModuleClusterModel{}
+}
+
+func (CloudHardwareSecurityModuleClusterResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return cloudhsmclusters.ValidateCloudHsmClusterID
+}
+
+func (r CloudHardwareSecurityModuleClusterResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.ValidateCloudHsmClusterName,
+		},
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"location": commonschema.Location(),
+
+		"auto_generated_domain_name_label_scope": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ForceNew:     true,
+			Default:      string(cloudhsmclusters.AutoGeneratedDomainNameLabelScopeTenantReuse),
+			ValidateFunc: validation.StringInSlice(cloudhsmclusters.PossibleValuesForAutoGeneratedDomainNameLabelScope(), false),
+		},
+
+		// only user assigned identity is supported even though the swagger shows system assigned identity
+		"identity": commonschema.UserAssignedIdentityOptional(),
+
+		"tags": commonschema.Tags(),
+	}
+}
+
+func (r CloudHardwareSecurityModuleClusterResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"activation_state": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"hsms": {
+			Type:     pluginsdk.TypeList,
+			Computed: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"fqdn": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+					"state": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+					"state_message": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+				},
+			},
+		},
+
+		"private_endpoint_connections": {
+			Type:     pluginsdk.TypeList,
+			Computed: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"id": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+					"name": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+					"type": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+					"group_ids": {
+						Type:     pluginsdk.TypeList,
+						Computed: true,
+						Elem: &pluginsdk.Schema{
+							Type: pluginsdk.TypeString,
+						},
+					},
+					"private_endpoint": {
```

Comment:
The naming is a bit redundent as the outside block is already named `private_endpoint_connections`.